airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/token_provider.py

```diff
@@ -62,7 +62,7 @@ def _refresh(self) -> None:
         )
         if response is None:
             raise ReadException("Failed to get session token, response got ignored by requester")
-        session_token = dpath.get(self._decoder.decode(response), self.session_token_path)
+        session_token = dpath.get(next(self._decoder.decode(response)), self.session_token_path)
```

Comment:
I believe this could fail if decode returns an empty iterable. can we catch the exception and raise a user-friendly error message if this happens?

Reply:
@girarda ,

what do you mean by `empty iterable`?
[Right now we catch](https://github.com/airbytehq/airbyte/pull/38829/files#diff-404f304b08b4a327259c7999f75283e1f111edaac2bb09196cb2c536f141f2deL23-L24) `except requests.exceptions.JSONDecodeError` and return empty dict `{}`

> can we catch the exception and raise a user-friendly error message if this happens?

Would you like to catch `JSONDecodeError`||`StreamConsumedError`||`ChunkedEncodingError`... or any exception of type `RequestException` and raise `AirbyteTracedException` of type `ConfigError` here?